File: Management/Models/Local/MarkdownUtils.cs
using System.Text.RegularExpressions;

namespace LocalModels;

public static class MarkdownUtils
{
  public static string ExtractLabelValue(string input, string label)
  {
    string pattern = $@"{label}: (.*?)\n";
    Match match = Regex.Match(input, pattern);

    if (match.Success)
    {
      return match.Groups[1].Value;
    }

    return string.Empty;
  }

}

File: src/models/local/assignment/utils/markdownUtils.ts
export const extractLabelValue = (input: string, label: string) => {
  const pattern = new RegExp(`${label}: (.*?)\n`);
  const match = pattern.exec(input);

  if (match && match.length > 1 && match[1]) {
    return match[1].trim();
  }

  return "";
};
